op-batcher/batcher/driver.go

```diff
@@ -272,10 +294,21 @@ func (l *BatchSubmitter) loop() {
 	// start the receipt/result processing loop
 	receiptLoopDone := make(chan struct{})
 	defer close(receiptLoopDone) // shut down receipt loop
+
+	var txpoolState atomic.Int32
+	txpoolState.Store(TxpoolGood)
 	go func() {
 		for {
 			select {
 			case r := <-receiptsCh:
+				if errors.Is(r.Err, txpool.ErrAlreadyReserved) && txpoolState.CompareAndSwap(TxpoolGood, TxpoolBlocked) {
+					l.Log.Info("incompatible tx in txpool")
```

Comment:
Why not just send the cancellation at this point, rather than waiting for another cycle through the loop? It goes from Good->Blocked, and then from Blocked->Pending, when it could maybe just go from Good->Pending?

Reply:
I thought it would be best to keep this loop purely handling receipts, and the other goroutine purely for sending. Mixing these responsibilities makes me a little nervous though I suppose it could work?  There could be potential blocking going on in the tx publishing routines, e.g. if queues get full for example, that might block receipt processing?

Reply:
That's a fair point, the status chain helps to keep the threads isolated by purpose :+1: